pkg/chartutil/create.go

```diff
@@ -516,6 +536,24 @@ Create the name of the service account to use
 {{- default "default" .Values.serviceAccount.name }}
 {{- end }}
 {{- end }}
+
+{{/*
+Create a checksum for all env parameters in the secret.
+Adds the checksum to podAnnotations to ensure pod reloads on env secret changes.
+*/}}
+{{- define "<CHARTNAME>.podAnnotations" -}}
+{{- if .Values.env -}}
+{{- $allKsAndVs := list -}}
```

Comment:
```suggestion
{{- $envChecksum := .Values.env | toYaml | sha256sum -}}
```

Reply:
good idea. i thought this might not work for multiline strings. i will try this